bundles/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/internal/OceanicHandlerFactory.java

```diff
@@ -40,17 +44,24 @@ public class OceanicHandlerFactory extends BaseThingHandlerFactory {
     private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
             .unmodifiableSet(Stream.of(THING_TYPE_SERIAL, THING_TYPE_NETWORK).collect(Collectors.toSet()));
 
+    private final SerialPortManager serialPortManager;
+
+    @Activate
+    public OceanicHandlerFactory(final @Reference SerialPortManager serialPortManager) {
+        this.serialPortManager = serialPortManager;
+    }
+
     @Override
-    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+    public boolean supportsThingType(@NonNull ThingTypeUID thingTypeUID) {
```

Comment:
The `@NonNull` annotation should not be used, see:

https://www.openhab.org/docs/developer/guidelines.html#null-annotations

```suggestion
    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
```